qadence/noise/protocols.py

```diff
@@ -1,29 +1,37 @@
 from __future__ import annotations
 
 import importlib
-from dataclasses import dataclass
-from typing import Callable, Counter, cast
+from typing import Any, Callable, Counter, cast
 
-from pyqtorch.noise import NoiseProtocol
+from qadence.types import NoiseEnum, NoiseProtocol
 
 PROTOCOL_TO_MODULE = {
-    "readout": "qadence.noise.readout",
+    "Readout": "qadence.noise.readout",
 }
 
-# Temporary solution
-DigitalNoise = NoiseProtocol
 
+class NoiseSource:
```

Comment:
I have the feeling all the checking `NoiseSource` provides can be done in the handler. Is that true ?

Reply:
I wanted to separate the checking of options at the NoiseSource level, and for the handler to check more the combination of several types.

Reply:
I get it, now it's removed.